carbon-design-system/carbon-platform
# feat: mdx-sanitizer

Closes #996
Closes #752
Closes #1105
Closes #1101 

MDX Sanitizer 🎉 👯 

Design Specs: https://www.figma.com/file/WyfqQh9R1VXYZErfmsNDVH/Platform_MDXerrors?node-id=0%3A1

Pending design:
- [ ] rollup notification (need #1087 )
- [ ] Common Errors page (need #1123 )


#### Changelog

**New**

- mdx-sanitizer
- mdx-processor
- `MdxPage` component: Replaces MdxWrapper and RemoteMdxLoader
- utils/mdx.js: creates mdx-processor and processes a given input source

**Changed**

- Dynamic Pages refactor: moved all pages into single [...remote-mdx-path] page
- update cached files

**Removed**

- All individual remote mdx pages

#### Testing / reviewing

- Pull from branch
- `npm install`
- run web-app
- visit http://localhost:3000/samples/remote-mdx-test. This will render mdx from https://raw.githubusercontent.com/francinelucca/mdx-testing/main/too-many-errors.mdx
     - Note rollup notification on the top with error summary (this is a temporary fix and will be replaced with proper design once #1087 is complete)
     - Confirm script tags get replaced with inlineError + Code snippet according to design specs
     - Note UnknownComponents get replaced with inlineError according to design specs
     * Hrefs for UnknownComponents don't work atm since storybook hasn't been deployed yet
- Replace line 44 in `services/web-app/pages/samples/remote-mdx-test.js` with `'with-imports.mdx'` ; this will render mdx from https://raw.githubusercontent.com/francinelucca/mdx-testing/main/with-imports.mdx
     - Note fullPageError is rendered according to design  specs
     - Confirm href links out to common errors page (this is a temporary fix and will be replaced with proper design once #1123  is complete)
- Replace line 44 in `services/web-app/pages/samples/remote-mdx-test.js` with `'with-exports.mdx'` ; this will render mdx from https://raw.githubusercontent.com/francinelucca/mdx-testing/main/with-exports.mdx
     - Note fullPageError is rendered according to design  specs
     - Confirm href links out to common errors page (this is a temporary fix and will be replaced with proper design once #1123  is complete)
- Replace line 44 in `services/web-app/pages/samples/remote-mdx-test.js` with `'with-comments.mdx'` ; this will render mdx from https://raw.githubusercontent.com/francinelucca/mdx-testing/main/with-comments.mdx
     - Note content is still able to render
- Replace line 44 in `services/web-app/pages/samples/remote-mdx-test.js` with `'parse-error.mdx'` ; this will render mdx from https://raw.githubusercontent.com/francinelucca/mdx-testing/main/parse-error.mdx
     - Confirm app does not error out and displays "Error compiling MDX" error instead as per design specs
     - Confirm href links out to common errors page (this is a temporary fix and will be replaced with proper design once #1123  is complete)
- Replace line 44 in `services/web-app/pages/samples/remote-mdx-test.js` with `'jsx-error.mdx'` ; this will render mdx from https://raw.githubusercontent.com/francinelucca/mdx-testing/main/jsx-error.mdx
     - Confirm app does not error out and displays "Error compiling MDX" error instead as per design specs
     * Hrefs for this type of error don't work atm since I'm not sure where they're supposed to link out to 🤔 . I've asked here: https://www.figma.com/file/WyfqQh9R1VXYZErfmsNDVH?node-id=9:4232#248066893
- Replace line 44 in `services/web-app/pages/samples/remote-mdx-test.js` with `'anything.mdx'` ; this will result in a content not found exception
     - Confirm app does not error out and displays "Content not found" fullPageError instead as per design specs
     - Confirm href links out to common errors page (this is a temporary fix and will be replaced with proper design once #1123  is complete)
 - This PR changes how dynamic remote pages are loaded, navigate as much as possible around the app and confirm everything renders and looks as expected, should have no regressions; also test library dynamic pages (http://localhost:3000/libraries/carbon-react/latest/pages/tutorial/overview)
